src/infra/flow-system/postgres/src/postgres_flow_event_store.rs

```diff
@@ -488,6 +496,61 @@ impl FlowEventStore for PostgresFlowEventStore {
         })
     }
 
+    /// Returns nearest flow activation time from all enqueued flows
+    async fn nearest_enqueued_moment(&self) -> Result<Option<DateTime<Utc>>, InternalError> {
+        let mut tr = self.transaction.lock().await;
+
+        let connection_mut = tr.connection_mut().await?;
+        let maybe_activation_time = sqlx::query!(
+            r#"
+            SELECT f.enqueued_for as activation_time
+                FROM flows f
+                WHERE
+                    f.enqueued_for IS NOT NULL AND
+                    f.flow_status = 'waiting'::flow_status_type
+                ORDER BY f.enqueued_for ASC
+                LIMIT 1
+            "#,
+        )
+        .map(|result| {
+            result
+                .activation_time
+                .expect("NULL values filtered by query")
+        })
+        .fetch_optional(connection_mut)
+        .await
+        .int_err()?;
+
+        Ok(maybe_activation_time)
+    }
+
+    /// Returns flows enqueued for the given activation time
+    async fn get_enqueued_flows(
+        &self,
+        enqueued_for: DateTime<Utc>,
+    ) -> Result<Vec<FlowID>, InternalError> {
+        let mut tr = self.transaction.lock().await;
+
+        let connection_mut = tr.connection_mut().await?;
+        let flow_ids = sqlx::query!(
+            r#"
+            SELECT f.flow_id as flow_id
+                FROM flows f
+                WHERE
+                    f.enqueued_for = $1 AND
```

Comment:
Hm, do we really want to filter by equality here? I expected to see a range query.

If the service was down for a while and then was started, don't we want to grab all `waiting` flows in one request and kick off their tasks, instead of doing potentially tens of thousands individual requests hopping a few milliseconds at a time?

Reply:
We are intentionally rounding activation times very carefully within the flow system, so this comparison is quite valid.
This was already necessary to support repeatable test patterns.